types/node/timers.d.ts

```diff
@@ -40,18 +40,31 @@ declare module 'timers' {
         }
 
         function setTimeout<TArgs extends any[]>(callback: (...args: TArgs) => void, ms?: number, ...args: TArgs): NodeJS.Timeout;
+        // Overload to allow setTimeout to take a Promise resolve callback
+        // with unspecified value type as is common when using `new Promise()`,
+        // e.g. new Promise(resolve => setInterval(resolve, 100));
+        // tslint:disable-next-line void-return
+        function setTimeout(callback: (args: void) => void): NodeJS.Timeout;
         namespace setTimeout {
             const __promisify__: typeof setTimeoutPromise;
         }
         function clearTimeout(timeoutId: NodeJS.Timeout): void;
 
         function setInterval<TArgs extends any[]>(callback: (...args: TArgs) => void, ms?: number, ...args: TArgs): NodeJS.Timer;
+        // As with setTimeout, overload to allow setInterval to take a Promise
```

Comment:
This is excessive, can be reduced to `// util.promisify no rest args compability`

Reply:
Sure!